packages/babel-parser/src/parser/expression.js

```diff
@@ -704,6 +699,71 @@ export default class ExpressionParser extends LValParser {
     }
   }
 
+  checkSliceArgument(pos: number, expr: N.Expression) {
+    if (
+      !expr ||
+      expr.type === "NumericLiteral" ||
+      expr.type === "Identifier" ||
```

Comment:
It seems super weird that Identifier is allowed, but normal expressions are not. 😞

Reply:
Actually I like this design choice. If nested general expression is supported, we can expect pretty nested code and one would easily get confused with ternary operator. Think of

```js
a[
 foo : (
   get_bar_from_qux()
 ) : (
  get_qux_from_quux()
 )
]
```
Surely we can open an issue on [the proposal](https://github.com/tc39/proposal-slice-notation#can-the-upper-bound-lower-bound-or-the-step-argument-be-an-arbitrary-expression) to discuss this choice.